test/functional/bipdersig-p2p.py

```diff
@@ -46,12 +46,8 @@ class BIP66Test(ComparisonTestFramework):
     def __init__(self):
         super().__init__()
         self.num_nodes = 1
-
-    def setup_network(self):
         # Must set the blockversion for this test
-        self.nodes = start_nodes(self.num_nodes, self.options.tmpdir,
-                                 extra_args=[['-whitelist=127.0.0.1', '-blockversion=2']],
-                                 binary=[self.options.testbinary])
+        self.extra_args = [[], ['-whitelist=127.0.0.1', '-blockversion=2']]
```

Comment:
This does not work for the `ComparisionFramework`, as it is later overwritten. I don't understand why this commit does not fail, but I'd still like the bug fixed in this pull.


Additionally, I'd like to see a sanity check in start_nodes:

```diff
diff --git a/test/functional/test_framework/util.py b/test/functional/test_framework/util.py
index 7730e14..9186c3c 100644
--- a/test/functional/test_framework/util.py
+++ b/test/functional/test_framework/util.py
@@ -354,6 +354,8 @@ def start_nodes(num_nodes, dirname, extra_args=None, rpchost=None, timewait=None
     """
     if extra_args is None: extra_args = [ None for _ in range(num_nodes) ]
     if binary is None: binary = [ None for _ in range(num_nodes) ]
+    assert_equal(len(extra_args), num_nodes)
+    assert_equal(len(binary), num_nodes)
     rpcs = []
     try:
         for i in range(num_nodes):